packages/e2e-tests/specs/editor/plugins/wp-editor-meta-box.test.js

```diff
@@ -39,8 +39,6 @@ describe( 'WP Editor Meta Boxes', () => {
 			( textarea ) => textarea.value
 		);
 
-		expect( content ).toMatchInlineSnapshot(
-			`"<p>Typing in a metabox</p>"`
-		);
+		expect( content ).toBe( 'Typing in a metabox' );
```

Comment:
Is this change expected, I feel this test has been updated back and forth several times already in the last couple releases, do we understand why this change is needed?

Reply:
It turns out that TinyMCE's WordPress plugin does the following:
https://github.com/WordPress/wordpress-develop/blob/2382765afa36e10bf3c74420024ad4e85763a47c/src/js/_enqueues/vendor/tinymce/plugins/wordpress/plugin.js#L15
https://github.com/WordPress/wordpress-develop/blob/2382765afa36e10bf3c74420024ad4e85763a47c/src/js/_enqueues/vendor/tinymce/plugins/wordpress/plugin.js#L132-L133
https://github.com/WordPress/wordpress-develop/blob/2382765afa36e10bf3c74420024ad4e85763a47c/src/js/_enqueues/vendor/tinymce/plugins/wordpress/plugin.js#L593-L609

```js
hasWpautop = ( wp && wp.editor && wp.editor.autop && editor.getParam( 'wpautop', true ) ),
editor.on( 'BeforeSetContent', function( event ) {
	// ...
	if ( hasWpautop ) {
	    event.content = wp.editor.autop( event.content );
	} else {
	    // Prevent creation of paragraphs out of multiple HTML comments
	    event.content = event.content.replace( /-->\s+<!--/g, '--><!--' );
	}
});

// ...

editor.on( 'SaveContent', function( event ) {
	// ...
	// Keep empty paragraphs :(
	event.content = event.content.replace( /<p>(?:<br ?\/?>|\u00a0|\uFEFF| )*<\/p>/g, '<p>&nbsp;</p>' );

	if ( hasWpautop ) {
		event.content = wp.editor.removep( event.content );
	} else {
		// Restore formatting of block boundaries.
		event.content = event.content.replace( /-->\s*<!-- wp:/g, '-->\n\n<!-- wp:' );
	}
});
```



So:
- if I call Object.assign( window.wp.editor, window.wp.oldEditor );  then hasWpauto is set to true by editor.getParam( 'wpautop', true ); .
	- When I switch from HTML to visual mode of TinyMCE, event.content is wrapped in a <p> 
	- When I switch back from the visual mode to the HTML mode, SaveContent is dispatched and it calls wp.editor.removep which removes the <p> tags.
- If I don't call the Object.assign thing, then hasWpauto is set to false. 
	- When I switch from HTML to visual mode of TinyMCE, event.content is not wrapped in a <p> but TinyMCE wraps it anyway.
	- When I switch back from the visual mode to the HTML mode, SaveContent is dispatched and it does not call wp.editor.removep so the <p> tags are not automatically removed at that stage.

Now, for the purposes of this test, we explicitly set wpautop to true in the test plugin here: https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/e2e-tests/plugins/wp-editor-metabox.php#L36 

This means that we do not want to see the the <p> tags in the editor. I added some documentation to that test to explain how to proceed in case it flips back again at some point @youknowriad .

Reply:
Thanks for the detailed explanation :)

Reply:
Good one @adamziel, this test should be finally reliable 👍🏻